entity-framework/core/managing-schemas/migrations/managing.md

```diff
@@ -0,0 +1,199 @@
+```
+
+***
+
+## Customize migration code
+
+While EF Core generally creates accurate migrations, you should always review the code and make sure it corresponds to the desired change; in some cases, it is even necessary to do so.
+
+### Column renames
+
+One notable examine where customizing migrations is required is when renaming a property. For example, if you rename a property from `Name` to `FullName`, EF Core will generate the following migration:
+
+```c#
+migrationBuilder.DropColumn(
+    name: "Name",
+    table: "Customers");
+
+migrationBuilder.AddColumn<string>(
+    name: "FullName",
+    table: "Customers",
+    nullable: true);
+```
+
+EF Core is generally unable to know when the intention is to drop a column and create a new one (two separate changes), and when a column should be renamed. If the above migration is applied as-is, all your customer names will be lost. To rename a column, replace the above generated migration with the following:
+
+```c#
+migrationBuilder.RenameColumn(
+    name: "Name",
+    table: "Customers",
+    newName: "FullName");
+```
+
+> [!TIP]
+> The migration scaffolding process warns when an operation might result in data loss (like dropping a column). If you see that warning, be especially sure to review the migrations code for accuracy.
+
+### Adding raw SQL
+
+While renaming a column can be achieved via a built-in API, in many cases that is not possible. For example, we may want to replace existing `FirstName` and `LastColumn` properties with a single, new `FullName` property. The migration generated by EF Core will be the following:
+
+``` csharp
+migrationBuilder.DropColumn(
+    name: "FirstName",
+    table: "Customer");
+
+migrationBuilder.DropColumn(
+    name: "LastName",
+    table: "Customer");
+
+migrationBuilder.AddColumn<string>(
+    name: "Name",
+    table: "Customer",
+    nullable: true);
+```
+
+As before, this would cause unwanted data loss. To transfer the data from the old columns, we rearrange the migrations and introduce a raw SQL operation as follows:
+
+``` csharp
+migrationBuilder.AddColumn<string>(
+    name: "Name",
+    table: "Customer",
+    nullable: true);
+
+migrationBuilder.Sql(
+@"
+    UPDATE Customer
+    SET Name = FirstName + ' ' + LastName;
+");
+
+migrationBuilder.DropColumn(
+    name: "FirstName",
+    table: "Customer");
+
+migrationBuilder.DropColumn(
+    name: "LastName",
+    table: "Customer");
+```
+
+### Arbitrary changes via raw SQL
+
+Raw SQL can also be used to manage database objects that EF Core isn't aware of. To do this, add a migration without making any model change; an empty migration will be generated, which you can then populate with raw SQL operations.
+
+For example, the following migration creates a SQL Server stored procedure:
+
+```c#
+migrationBuilder.Sql(
+@"
+    CREATE PROCEDURE getFullName
+        @LastName nvarchar(50),
+        @FirstName nvarchar(50)
+    AS
+        RETURN @LastName + @FirstName;
```

Comment:
We should probably wrap this inside a T-SQL `EXEC` call so it works with idempotent scripts.

Reply:
Good catch, thanks - did not know about this. I guess this is another minor caveat when using idempotent migrations.